corehq/apps/hqadmin/forms.py

```diff
@@ -114,6 +96,81 @@ def __init__(self, can_toggle_is_staff, *args, **kwargs):
         )
 
 
+class OffboardingUserListForm(forms.Form):
+    csv_email_list = forms.CharField(
+        label="Comma seperated email addresses",
+        widget=forms.Textarea()
+    )
+    invert_apply = forms.BooleanField(
+        label="Invert",
+        initial=False,
+        required=False,
+        help_text=(
+            "Get dimagi.com accounts not included in the above, comma-separated list."
+        )
+    )
+
+    def clean(self):
+        invert = self.cleaned_data.get('invert_apply', False)
+        return clean_data(self.cleaned_data, invert=invert, offboarding_list=True)
+
+    def __init__(self, *args, **kwargs):
+        super(OffboardingUserListForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = "form-horizontal"
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.layout = crispy.Layout(
+            'csv_email_list',
+            'invert_apply',
+            FormActions(
+                crispy.Submit(
+                    'superuser_management',
+                    'Get Users'
+                )
+            )
+        )
+
+
+def clean_data(cleaned_data, invert=False, offboarding_list=False):
+    EMAIL_INDEX = 1
+    csv_email_list = cleaned_data.get('csv_email_list', '')
+    csv_email_list = re.split(',|\n', csv_email_list)
+    csv_email_list = [parseaddr(email)[EMAIL_INDEX] for email in csv_email_list]
+
+    MAX_ALLOWED_EMAIL_USERS = 10
+    if not offboarding_list and len(csv_email_list) > MAX_ALLOWED_EMAIL_USERS:
+        raise forms.ValidationError(
+            f"This command allows superusers to modify up to {MAX_ALLOWED_EMAIL_USERS} users at a time. "
+            "If you are trying to update permissions for a larger number of users,"
+            " consider doing it via Django Admin"
+        )
+
+    users = []
+    validationErrors = []
+    for username in csv_email_list:
+        if not offboarding_list:
+            if settings.IS_DIMAGI_ENVIRONMENT and not is_dimagi_email(username):
+                validationErrors.append(ValidationError(_("Email address '{}' is not a "
+                                                          "dimagi email address".format(username))))
+                continue
+        try:
+            users.append(User.objects.get(username=username))
+        except User.DoesNotExist:
+            validationErrors.append(ValidationError(_(
+                "User with email address '{}' does not exist on "
+                "this site, please have the user registered first".format(username))))
+    if validationErrors:
+        raise ValidationError(validationErrors)
+
+    if invert:
+        all_users = User.objects.filter(Q(is_superuser=True) | Q(is_staff=True))
```

Comment:
I think we want this query to find all users to whom any of the following apply:
- is_superuser=True
- is_staff=True
- they're a "Dimagi" user (the username ends in @dimagi.com) and they're active